### PR TITLE
[FIL-245] Keep unsealed

### PR DESCRIPTION
**Pull request by ortyomka**
### Description of the Change
- Added Keep Unsealed
- Added Sector File
- Change Rust FFI writer to Cpp Writer

### Benefits
- Optimize reading
- Optimize unsealing
- Our own implementation

### Possible Drawbacks 
- Maybe slower than Rust

### CODE REVIEW

File: core/primitives/bitvec/bitvec.hpp [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    /**
+     * Flush content to vector
+     * @return bitvec vector
+     */
+    std::vector<uint8_t> out() {
+      if (bitsCap != 0) {
+        buffer.push_back(uint8_t(bits));
+      }
+      if (bitsCap > 8) {
+        buffer.push_back(uint8_t(uint8_t(bits >> 8)));
```
Comment (turuslan) [inline core/primitives/bitvec/bitvec.hpp:123]: Double cast?

File: core/primitives/bitvec/bitvec.hpp [hunk 2]
```diff
@@ -0,0 +1,164 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef CPP_FILECOIN_CORE_PRIMITIVES_RLE_BITSET_RLE_BITVEC_HPP
```
Comment (Alexey-N-Chernyshov) [inline core/primitives/bitvec/bitvec.hpp:6]: Let's use `#pragma once`

File: core/primitives/rle_bitset/runs_utils.hpp [hunk 3]
```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef CPP_FILECOIN_CORE_PRIMITIVES_RLE_BITSET_RUNS_UTILS_HPP
```
Comment (Alexey-N-Chernyshov) [inline core/primitives/rle_bitset/runs_utils.hpp:6]: `#pragma once`

File: test/core/primitives/bitvec/bitvec_test.cpp [hunk 4]
```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "primitives/bitvec/bitvec.hpp"
+
+#include <gtest/gtest.h>
+
```
Comment (Alexey-N-Chernyshov) [inline test/core/primitives/bitvec/bitvec_test.cpp:9]: Place tests in `namespace fc::primitives`

File: core/primitives/bitvec/bitvec.hpp [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+#define CPP_FILECOIN_CORE_PRIMITIVES_RLE_BITSET_RLE_BITVEC_HPP
+
+#include <cstdint>
+#include <vector>
+
+namespace fc::primitives {
```
Comment (Alexey-N-Chernyshov) [inline core/primitives/bitvec/bitvec.hpp:12]: Namespace and directory doesn't match, it is not required but recommended. Please consider putting `BitvecReader` class in `fc::primitives::bitvec`.

File: test/core/primitives/rle_bitset/runs_utils_test.cpp [hunk 6]
```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "primitives/rle_bitset/runs_utils.hpp"
+
+#include <gtest/gtest.h>
+
+TEST(RunsUtils, OrSuccess) {
```
Comment (Alexey-N-Chernyshov) [inline test/core/primitives/rle_bitset/runs_utils_test.cpp:10]: namespace

Comment (Alexey-N-Chernyshov) [inline test/core/primitives/rle_bitset/runs_utils_test.cpp:10]: Please, provide short description of tests.

File: test/core/primitives/sector_file/sector_file_test.cpp [hunk 7]
```diff
@@ -4,23 +4,239 @@
...
 #include "primitives/sector_file/sector_file.hpp"
-#include <gtest/gtest.h>
 
+#include <gtest/gtest.h>
+#include <common/file.hpp>
 #include "testutil/outcome.hpp"
+#include "testutil/resources/resources.hpp"
+#include "testutil/storage/base_fs_test.hpp"
+
+using fc::primitives::sector_file::kOverheadDenominator;
```
Comment (Alexey-N-Chernyshov) [inline test/core/primitives/sector_file/sector_file_test.cpp:14]: Consider putting tests into `fc::primitives::sector_file` namespace.

**Review by turuslan — APPROVED** [review]

File: test/core/primitives/sector_file/sector_file_test.cpp (hunk 7 above)
Comment (ortyomka) [reply test/core/primitives/sector_file/sector_file_test.cpp]: Fixed

File: test/core/primitives/rle_bitset/runs_utils_test.cpp (hunk 6 above)
Comment (ortyomka) [reply test/core/primitives/rle_bitset/runs_utils_test.cpp]: Fixed

Comment (ortyomka) [reply test/core/primitives/rle_bitset/runs_utils_test.cpp]: Fixed

File: core/primitives/bitvec/bitvec.hpp (hunk 5 above)
Comment (ortyomka) [reply core/primitives/bitvec/bitvec.hpp]: Fixed

File: test/core/primitives/bitvec/bitvec_test.cpp (hunk 4 above)
Comment (ortyomka) [reply test/core/primitives/bitvec/bitvec_test.cpp:9]: Fixed

File: core/primitives/rle_bitset/runs_utils.hpp (hunk 3 above)
Comment (ortyomka) [reply core/primitives/rle_bitset/runs_utils.hpp]: Fixed

File: core/primitives/bitvec/bitvec.hpp (hunk 2 above)
Comment (ortyomka) [reply core/primitives/bitvec/bitvec.hpp]: Fixed

File: core/primitives/bitvec/bitvec.hpp (hunk 1 above)
Comment (ortyomka) [reply core/primitives/bitvec/bitvec.hpp]: Fixed

File: core/primitives/sector_file/sector_file.hpp [hunk 8]
```diff
@@ -6,10 +6,20 @@
 #ifndef CPP_FILECOIN_SECTOR_FILE_HPP
```
Comment (Alexey-N-Chernyshov) [inline core/primitives/sector_file/sector_file.hpp:6]: #pragma once

**Review by Alexey-N-Chernyshov — APPROVED** [review]

**Pull request merged**